audio/audio_hw.c

```diff
@@ -396,9 +398,84 @@ static int out_set_parameters(struct audio_stream *stream, const char *kvpairs)
     return 0;
 }
 
-static char * out_get_parameters(const struct audio_stream *stream, const char *keys)
+static void str_parms_add_audio_format(struct str_parms* str_parms, 
+    const char* audio_parms, audio_format_t audio_format) {
+    char* format;
+
+    switch (audio_format) {
```

Comment:
Should be audio_get_main_format(audio_format)

Reply:
I have tried to use this function, but the VTS test did not pass after that.
https://android.googlesource.com/platform/hardware/interfaces/+/refs/tags/android-10.0.0_r17/audio/core/all-versions/vts/functional/AudioPrimaryHidlHalTest.h#858

Reply:
Yes, you're right. I didn't notice that your switch values are main+sub so it needs the full value.